### PR TITLE
Add eu.ad5001.LogarithmPlotter

### PR DESCRIPTION
**Pull request by Ad5001**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - I'm using `--filesystem=home` for LogarithmPlotter to be able to open the user's files from their home directory.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.. **Link:** https://git.ad5001.eu/Ad5001/LogarithmPlotter ([mirror on Github](https://github.com/Ad5001/LogarithmPlotter))
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
     - N/A

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: eu.ad5001.LogarithmPlotter.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "finish-args": [
+        "--share=ipc",
+        "--device=dri",
+        "--socket=x11",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.DBus.Proprieties",
+        "--talk-name=org.freedesktop.IBus",
+        "--talk-name=org.freedestkop.portal.*",
```
Comment (barthalion) [inline eu.ad5001.LogarithmPlotter.json:13]: What is it needed for?

File: eu.ad5001.LogarithmPlotter.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "cleanup": [ ],
+    "post-install": [
+        "install -D mimetypes /app/share/mime/packages/x-logarithm-plot.xml"
+    ],
+    "modules": [
+        "python3-pyside2.json",
+        {
+            "name": "LogarithmPlotter",
+            "buildsystem": "simple",
+            "config-opts": ["no-debuginfo-compression"],
```
Comment (barthalion) [inline eu.ad5001.LogarithmPlotter.json:29]: What is it needed for?

File: eu.ad5001.LogarithmPlotter.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        "--talk-name=org.freedesktop.DBus.Proprieties",
+        "--talk-name=org.freedesktop.IBus",
+        "--talk-name=org.freedestkop.portal.*",
+        "--filesystem=home"
+    ],
+    "command": "logarithmplotter",
+    "rename-desktop-file": "logarithmplotter.desktop",
+    "rename-icon": "logplotter",
+    "rename-appdata-file": "eu.ad5001.LogarithmPlotter.metainfo.xml",
+    "cleanup": [ ],
```
Comment (barthalion) [inline eu.ad5001.LogarithmPlotter.json:20]:
```suggestion
```

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{ 
+    "only-arches": ["x86_64"]
```
Comment (barthalion) [inline flathub.json:2]: Any reason to skip aarch64?

File: eu.ad5001.LogarithmPlotter.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    "post-install": [
+        "install -D mimetypes /app/share/mime/packages/x-logarithm-plot.xml"
+    ],
+    "modules": [
+        "python3-pyside2.json",
```
Comment (barthalion) [inline eu.ad5001.LogarithmPlotter.json:25]: It's preferred to keep all modules a single manifest. You can pretty much copy the PySide2 module from that file here.

File: eu.ad5001.LogarithmPlotter.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{
+    "app-id": "eu.ad5001.LogarithmPlotter",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "finish-args": [
+        "--share=ipc",
+        "--device=dri",
+        "--socket=x11",
```
Comment (barthalion) [inline eu.ad5001.LogarithmPlotter.json:9]:
```suggestion
        "--socket=fallback-x11",
```

File: eu.ad5001.LogarithmPlotter.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+    "app-id": "eu.ad5001.LogarithmPlotter",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
```
Comment (barthalion) [inline eu.ad5001.LogarithmPlotter.json:4]:
```suggestion
    "runtime-version": "5.15-21.08",
```

File: eu.ad5001.LogarithmPlotter.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        "--talk-name=org.freedestkop.portal.*",
+        "--filesystem=home"
+    ],
+    "command": "logarithmplotter",
+    "rename-desktop-file": "logarithmplotter.desktop",
+    "rename-icon": "logplotter",
+    "rename-appdata-file": "eu.ad5001.LogarithmPlotter.metainfo.xml",
+    "cleanup": [ ],
+    "post-install": [
+        "install -D mimetypes /app/share/mime/packages/x-logarithm-plot.xml"
```
Comment (hfiguiere) [inline eu.ad5001.LogarithmPlotter.json:22]: I don't think this works.
1. there is no `mimetypes` at the top level directory of the source tree.
2. the mimetype file should be name `${FLATPAK_ID}.xml`

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
review streams crossed...

File: eu.ad5001.LogarithmPlotter.json (hunk 2 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:29]: Oh this is a residue from when I tried another method to compile PySide2. I'll remove it.

File: eu.ad5001.LogarithmPlotter.json (hunk 5 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:25]: Oh really? I thought it was good practice to keep shared modules in separate files. Thanks for the advice.

Comment (hfiguiere) [reply eu.ad5001.LogarithmPlotter.json:25]: the only case is if you have a manifest generated by a tool. Or use "shared-modules".

File: flathub.json (hunk 4 above)
Comment (Ad5001) [reply flathub.json:2]: Originally PySIde2 was only compiled for x86_64, but you're right, there shouldn't be any reason it not being compilable on aarch64.

File: eu.ad5001.LogarithmPlotter.json (hunk 5 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:25]: Ah I see

File: eu.ad5001.LogarithmPlotter.json (hunk 8 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:22]: Is there any way to use something like "rename-mime-file" for this?

Comment (hfiguiere) [reply eu.ad5001.LogarithmPlotter.json:22]: sadly no.

File: eu.ad5001.LogarithmPlotter.json [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.DBus.Proprieties",
+        "--talk-name=org.freedesktop.IBus",
+        "--talk-name=org.freedestkop.portal.*",
+        "--filesystem=home"
+    ],
+    "command": "logarithmplotter",
+    "rename-desktop-file": "logarithmplotter.desktop",
+    "rename-icon": "logplotter",
+    "rename-appdata-file": "eu.ad5001.LogarithmPlotter.metainfo.xml",
```
[the remark this answers is not in the document]
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:19]: I thought the rename function was also used to point the file to flatpak-builder. Is it not necessary?

[the remark this answers is not in the document]
Comment (hfiguiere) [reply eu.ad5001.LogarithmPlotter.json:19]: it's used to rename. If the file has the right name, then there is nothing to do.

File: eu.ad5001.LogarithmPlotter.json (hunk 8 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:21]: I see, is it possible to just rename the mime file in the build command and flatpak-builder would detect it immediately ?

Comment (hfiguiere) [reply eu.ad5001.LogarithmPlotter.json:21]: you can use `mv` if you prefer (but if it's not installed, `install` is best), but the `post-install` should be located in the module.

File: flathub.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{ 
+    "only-arches": ["x86_64", "aarch64"]
```
Comment (hfiguiere) [inline flathub.json:2]: you can delete this file. These are the only architectures build.

File: eu.ad5001.LogarithmPlotter.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json]: Removed in newer commit

File: eu.ad5001.LogarithmPlotter.json (hunk 5 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json]: Fixed in latest commit.

File: eu.ad5001.LogarithmPlotter.json (hunk 2 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json]: Fixed in latest commit.

File: flathub.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ad5001) [reply flathub.json]: Fixed in latest commit.

Comment (Ad5001) [reply flathub.json]: Fixed in latest commit.

File: eu.ad5001.LogarithmPlotter.json (hunk 8 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json]: Fixed in latest commit.

**Comment by hfiguiere** [comment]
bot, build eu.ad5001.LogarithmPlotter

File: eu.ad5001.LogarithmPlotter.json (hunk 7 above)
Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json]: There seems to be an issue with building PySide2 under the KDE SDK & runtime v5.15-21.08 where CLang isn't being properly detected anymore. I'll revert back to v5.15 in the meantime. 
```
CMake Error at sources/shiboken2/data/shiboken_helpers.cmake:146 (message):
  Unable to detect CLANG location by checking LLVM_INSTALL_DIR,
  CLANG_INSTALL_DIR or running llvm-config.
Call Stack (most recent call first):
  sources/shiboken2/CMakeLists.txt:39 (setup_clang)
```

**Comment by Ad5001** [comment]
The issue the build just had was due to the version 5.15-21.08 of the KDE Runtime. I've reverted it in the latest commit so it shouldn't have any building issue anymore.

**Comment by Ad5001** [comment]
bot, build eu.ad5001.LogarithmPlotter

File: eu.ad5001.LogarithmPlotter.json [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    "app-id": "eu.ad5001.LogarithmPlotter",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "finish-args": [
+        "--share=ipc",
+        "--device=dri",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--talk-name=org.freedesktop.DBus.Proprieties",
```
Comment (nedrichards) [inline eu.ad5001.LogarithmPlotter.json:11]: I think you mean properties https://dbus.freedesktop.org/doc/dbus-specification.html#standard-interfaces-properties

Comment (Ad5001) [reply eu.ad5001.LogarithmPlotter.json:11]: Can't believe I missed that! Nice catch.

**Comment by Ad5001** [comment]
Queued test build for eu.ad5001.LogarithmPlotter.

**Comment by Ad5001** [comment]
bot, build eu.ad5001.LogarithmPlotter

File: eu.ad5001.LogarithmPlotter.json (hunk 1 above)
Comment (nedrichards) [reply eu.ad5001.LogarithmPlotter.json:13]: I'm slightly confused as to why you need to talk to the portal for open/save but also have the --filesystem=home permission.

**Comment by barthalion** [comment]
bot, build eu.ad5001.LogarithmPlotter

**Comment by Ad5001** [comment]
Just did a few checks, all functionalities seem to work without the permissions you've removed. Thanks @barthalion!
Is there any other requirement to be fulfilled before merging?

**Comment by barthalion** [comment]
Nope, I've been going through submissions and noticed yours fell of a radar.

One thing that could be improved long term is using portals to open and save files: https://docs.flatpak.org/en/latest/portals-qt.html

…it's not a blocker though.

/merge

**Pull request closed without merging**